src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodContainerSourceTest.java

```diff
@@ -0,0 +1,96 @@
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.WithoutJenkins;
+
+public class PodContainerSourceTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void lookupContainerWorkingDir() {
+        Pod pod = new PodBuilder()
+                .withNewSpec()
+                .addNewContainer()
+                .withName("foo")
+                .withWorkingDir("/app/foo")
+                .endContainer()
+                .addNewEphemeralContainer()
+                .withName("bar")
+                .withWorkingDir("/app/bar")
+                .endEphemeralContainer()
+                .addNewEphemeralContainer()
+                .withName("foo")
+                .withWorkingDir("/app/ephemeral-foo")
+                .endEphemeralContainer()
+                .endSpec()
+                .build();
+
+        Optional<String> wd = PodContainerSource.lookupContainerWorkingDir(pod, "foo");
+        assertTrue(wd.isPresent());
+        assertEquals("/app/foo", wd.get());
+
+        // should use TestPodContainerSource to find ephemeral container
+        wd = PodContainerSource.lookupContainerWorkingDir(pod, "bar");
+        assertTrue(wd.isPresent());
+        assertEquals("/app/bar", wd.get());
+
+        // no named container
+        wd = PodContainerSource.lookupContainerWorkingDir(pod, "fish");
+        assertFalse(wd.isPresent());
+    }
+
+    @WithoutJenkins
+    @Test
+    public void defaultPodContainerSourceGetContainerWorkingDir() {
+        Pod pod = new PodBuilder()
+                .withNewSpec()
+                .addNewContainer()
+                .withName("foo")
+                .withWorkingDir("/app/foo")
+                .endContainer()
+                .addNewEphemeralContainer()
+                .withName("bar")
+                .withWorkingDir("/app/bar")
+                .endEphemeralContainer()
+                .endSpec()
+                .build();
+
+        PodContainerSource.DefaultPodContainerSource source = new PodContainerSource.DefaultPodContainerSource();
+        Optional<String> wd = source.getContainerWorkingDir(pod, "foo");
+        assertTrue(wd.isPresent());
+        assertEquals("/app/foo", wd.get());
+
+        // should not return ephemeral container
+        wd = source.getContainerWorkingDir(pod, "bar");
+        assertFalse(wd.isPresent());
+
+        // no named container
+        wd = source.getContainerWorkingDir(pod, "fish");
+        assertFalse(wd.isPresent());
+    }
+
+    @Extension
+    public static class TestPodContainerSource extends PodContainerSource {
+
+        @Override
+        public Optional<String> getContainerWorkingDir(@NotNull Pod pod, @NotNull String containerName) {
```

Comment:
Use FindBugs/Spotbugs annotations instead (jetbrains annotations have been removed from the Jenkins parent pom, however it seems to be still present for tests through some transitive dependencies)

```suggestion
        public Optional<String> getContainerWorkingDir(@NonNull Pod pod, @NonNull String containerName) {
```

Reply:
https://github.com/jenkinsci/kubernetes-plugin/pull/1694